packages/sync/src/useSync.ts

```diff
@@ -275,4 +290,12 @@ export interface UseSyncOptions {
 	roomId?: string
 	/** @internal */
 	trackAnalyticsEvent?(name: string, data: { [key: string]: any }): void
+
+	/**
+	 * A reactive function that returns a {@link @tldraw/tlschema#TLInstancePresence} object. The
+	 * result of this function will be synchronized across all clients to display presence
+	 * indicators such as cursors. See {@link @tldraw/tlschema#getDefaultUserPresence} for
+	 * the default implementation of this function.
+	 */
+	getUserPresence?(store: TLStore, user: TLSyncUserInfo): TLPresenceStateInfo | null
```

Comment:
I think at the same time as doing this we should really make the cursor and camera properties of `TLInstancePresence` nullable, and to have default values of `null`. Otherwise there's no way to e.g. actually omit the user's cursor or camera positions and for us to handle that while rendering presence components. It just gets set to a meaningless default (0,0). 

Maybe the same for `currentPageId` and `screenBounds`?

Also if they forget to add a `lastActivityTimestamp` it will always be 0 and people's cursors will always be hidden by our UI at least. Maybe we should switch the default for that to `+Infinity` to make the cursors never disappear unless they give a `lastActivityTimestamp`?

I think the rest of the properties should be fine since their default states are also empty states.

Reply:
This makes sense. On `lastActivityTimestamp`, I think we might have issues as `Infinity` can't be serialized as JSON - it comes through as `null` instead. Maybe we treat `null` as `Infinity`?

Reply:
sounds good